### PR TITLE
Fix (potential) double-free in WaveformMarkRange

### PR DESCRIPTION
**Pull request by uklotzde**
...to prevent further damage and mistakes. #1242 needs to pick up these fixes!

Hopefully this bug has not been discovered yet, because the affected code paths are unused.

### CODE REVIEW

**Comment by ninomp** [comment]
Oh, so we don't need to define move constructor to fix this?

**Comment by uklotzde** [comment]
The implicitly defined move constructor is sufficient when using std::unique_ptr instead of plain pointers.

**Comment by daschuer** [comment]
The double free happen if you copy the WaveformMarkRange. 
I think we did not copy it so this is not a "bug" from the users perspective.

The commit acutually prevents the copy, which would inyroduce the bug.

Maybe we should reword the PR title and maybe also the commit message to not scare anyone.

What do you think?

Can you also delete the copy constructor and assignment operator to make it explicite.

The rest LGTM

**Comment by uklotzde** [comment]
Instances of this class are stored in a vector which uses the implicitly defined and broken copy constructor. I still don't understand why we don't already experience crashes.

An implicit move constructor should not have been generated, because of the explicit constructor and destructor.

**Comment by uklotzde** [comment]
The bug is not triggered, because the copy constructor is currently only invoked with a default constructed instance where all internal pointers are still *null*.

**Comment by uklotzde** [comment]
The setup() member function is completely unnecessary and WaveformMarkRange should instead be fully initialized by the constructor.

**Comment by uklotzde** [comment]
Instances of this class could be made immutable by declaring all members as `const`, but I'll stop at this point.

File: src/waveform/renderers/waveformmarkrange.h [hunk 1]
```diff
@@ -4,37 +4,38 @@
...
 
 class ControlProxy;
 class QDomNode;
 class WaveformSignalColors;
 
 class WaveformMarkRange {
   public:
-    WaveformMarkRange();
-    ~WaveformMarkRange();
+    WaveformMarkRange(const QString &group, const QDomNode& node,
```
Comment (ninomp) [inline src/waveform/renderers/waveformmarkrange.h]: Nitpick: Inconsistent alignment of `&`.

**Comment by ninomp** [comment]
LGTM

**Comment by daschuer** [comment]
https://github.com/mixxxdj/mixxx/blob/9da873a777a147a3afadfb3a988ec42112359d1e/src/waveform/renderers/waveformrendermarkrange.cpp#L31

This is the only place, we "copy" the WaveformMarkRange. In this case the pointers are still nullptr and can be safely deleted twice. The compiler actually optimizes the copy call away. 
In your case it uses an implicit move assignment.

**Comment by daschuer** [comment]
Thank you.

**Comment by Be-ing** [comment]
Why was this merged to 2.1? I thought we agreed 2.1.7 would be the last 2.1 release??

**Comment by daschuer** [comment]
I am not planning a 2.1.8 release.
This was just developed and tested on the 2.1 branch. I have also considered to merge this to master only, but I did not had a strong opinion about it.

**Comment by Be-ing** [comment]
Now we have the 2.1 branch in a state inconsistent with its last release. This is probably not a big deal, but it might confuse someone in the future.

**Comment by uklotzde** [comment]
I decided to target 2.1, because it was such a serious flaw that could blow up at any time. It doesn't hurt that the 2.1 branch is now ahead of the last release label.

**Pull request merged**